### PR TITLE
Add sas support to SB

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/14572

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/samples/sync_samples/authenticate_using_azure_sas_credential.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+auth_uri = "sb://{}/{}".format(FULLY_QUALIFIED_NAMESPACE, SESSION_QUEUE_NAME)
+token_ttl = 3000  # seconds
+
+sas_token = generate_sas_token(auth_uri, SAS_POLICY, SERVICEBUS_SAS_KEY, token_ttl)
+
+credential=AzureSasCredential(sas_token)
+
+with ServiceBusClient(FULLY_QUALIFIED_NAMESPACE, credential) as client:
+    with client.get_queue_sender(SESSION_QUEUE_NAME) as sender:
+        sender.send_messages([ServiceBusMessage("hello")])
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/samples/sync_samples/authenticate_using_azure_sas_credential.py:49]: I don't think we need a session queue, normal queue should be fine:
```python
QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
```
also we should revert the following:
```python
# SESSION_QUEUE_NAME = os.environ["SERVICE_BUS_SESSION_QUEUE_NAME"]
# SAS_POLICY = os.environ['SERVICE_BUS_SAS_POLICY']
SESSION_QUEUE_NAME = "sbsessionqueue"
SAS_POLICY = "test"
```

File: sdk/servicebus/azure-servicebus/samples/async_samples/authenticate_using_azure_sas_credential_async.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+auth_uri = "sb://{}/{}".format(FULLY_QUALIFIED_NAMESPACE, SESSION_QUEUE_NAME)
+token_ttl = 3000  # seconds
+
+sas_token = generate_sas_token(auth_uri, SAS_POLICY, SERVICEBUS_SAS_KEY, token_ttl)
+
+async def send_message():
+    credential=AzureSasCredential(sas_token)
+    async with ServiceBusClient(FULLY_QUALIFIED_NAMESPACE, credential) as client:
+        async with client.get_queue_sender(SESSION_QUEUE_NAME) as sender:
+            await sender.send_messages([ServiceBusMessage("hello")])
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/samples/async_samples/authenticate_using_azure_sas_credential_async.py:51]: I don't think we need a session queue, normal queue should be fine, as most users would use normal queue:
```python
QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
```
also we should revert the following:
```python
# SESSION_QUEUE_NAME = os.environ["SERVICE_BUS_SESSION_QUEUE_NAME"]
# SAS_POLICY = os.environ['SERVICE_BUS_SAS_POLICY']
SESSION_QUEUE_NAME = "sbsessionqueue"
SAS_POLICY = "test"
```

File: scripts/devops_tasks/test_run_samples.py [hunk 3]
```diff
@@ -99,7 +99,9 @@
         "mgmt_subscription_async.py",
         "mgmt_topic_async.py",
         "proxy_async.py",
-        "receive_deferred_message_queue_async.py"
+        "receive_deferred_message_queue_async.py",
+        "authenticate_using_azure_sas_credential_async.py",
+        "authenticate_using_azure_sas_credential.py"
```
Comment (yunhaoling) [inline scripts/devops_tasks/test_run_samples.py:104]: is there a reason to ignore those two? looking into the samples I think they're runnable

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py [hunk 4]
```diff
@@ -39,10 +39,11 @@ class ServiceBusClient(object):
 
     :param str fully_qualified_namespace: The fully qualified host name for the Service Bus namespace.
      The namespace format is: `<yournamespace>.servicebus.windows.net`.
-    :param ~azure.core.credentials.TokenCredential credential: The credential object used for authentication which
+    :param credential: The credential object used for authentication which
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:42]: changelog not updated :P

File: sdk/servicebus/azure-servicebus/samples/async_samples/authenticate_using_azure_sas_credential_async.py (hunk 2 above)
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/samples/async_samples/authenticate_using_azure_sas_credential_async.py:51]: I don't think we need a session queue, normal queue should be fine, as most users would use normal queue:
```python
QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
```
also we should revert the following:
```python
# SESSION_QUEUE_NAME = os.environ["SERVICE_BUS_SESSION_QUEUE_NAME"]
# SAS_POLICY = os.environ['SERVICE_BUS_SAS_POLICY']
SESSION_QUEUE_NAME = "sbsessionqueue"
SAS_POLICY = "test"
```

File: scripts/devops_tasks/test_run_samples.py (hunk 3 above)
Comment (yunhaoling) [reply scripts/devops_tasks/test_run_samples.py:104]: skipping it for now, env variables are not ready

Comment (yunhaoling) [reply scripts/devops_tasks/test_run_samples.py:104]: could we create an issue for tracking?

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,7 @@
 **New Features**
 
 * Updated `forward_to` and `forward_dead_lettered_messages_to` parameters in `create_queue`, `update_queue`, `create_subscription`, and `update_subscription` methods on sync and async `ServiceBusAdministrationClient` to accept entities as well, rather than only full paths. In the case that an entity is passed in, it is assumed that the entity exists within the same namespace used for constructing the `ServiceBusAdministrationClient`.
+* Added support for using `~azure.core.credentials.AzureSasCredential` as credential for authenticating the clients.
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:8]: haha, we should have a new entry # 7.2.0b1 to hold this new feature

Comment (rakshith91) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:8]: this the only feature in b1? :)

File: scripts/devops_tasks/test_run_samples.py (hunk 3 above)
Comment (swathipil) [reply scripts/devops_tasks/test_run_samples.py:104]: I can create one

File: sdk/servicebus/azure-servicebus/CHANGELOG.md (hunk 5 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:8]: nah, I have a pending sending amqp annotated message PR to be merged: https://github.com/Azure/azure-sdk-for-python/pull/17650

but I'll let yours get merged first and then start resolving conflicts if needed.

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**